### ISSUE TITLE
Performance with TPU vs Pytorch( GPU P100)

### ISSUE THREAD

**Issue by rashmibanthia** [google-research/bert#687]
I am trying a fine tuning task with Bert base. I have tried the same with pytorch pretrained (huggingface/pytorch-pretrained-BERT)  on P100. 

I noticed a significant improvement in speed but not in performance. I tried to keep the same code / settings, but unable to match or beat the performance (a variant of AUC).  I am happy to provide more details about the dataset/metric etc.    (I have a spreadsheet of different  batch sizes/ learning rate/ epochs/ warmup proportion that I have tried - ~20 combinations) 

(Evaluation set metric on Pytorch P100 = 0.9412374   vs. TPU = 0.9376376 

Any help/suggestions much appreciated.  Thank you.

```
#Loss function on Pytorch: 

def custom_loss(data, targets):  #data=predictions, target = groundtruth
    ''' Define custom loss function for weighted BCE on 'target' column '''
    bce_loss_1 = nn.BCEWithLogitsLoss(weight=targets[:,1:2])(data[:,:1],targets[:,:1]) #target
    bce_loss_2 = nn.BCEWithLogitsLoss()(data[:,2:],targets[:,2:]) #aux targets
    return (bce_loss_1 * loss_weight) + bce_loss_2

```

 
```
#Loss calculation on TF: 
per_example_loss1 = tf.nn.weighted_cross_entropy_with_logits(pos_weight=labels[:,1], targets=labels[:,0], logits=logits[:,0])
per_example_loss2 = tf.nn.sigmoid_cross_entropy_with_logits( labels=labels[:,2:], logits=logits[:,2:])
per_example_loss = (per_example_loss1 * loss_weight) + tf.reduce_mean(per_example_loss2,axis=1)
    
loss = tf.reduce_mean(per_example_loss)
```
